### PR TITLE
better hashing and graphql mutations

### PR DESCRIPTION
**Pull request by prairir**
### What are you trying to accomplish?
implement better password hashing

### How are you accomplishing it?
writing better hashing functions, there was a security issue that the password was saved to user and then hashed, if the hash failed then it would continue with the original plaintext password. thats a 1/10 issue so i removed it.

### Is there anything reviewers should know?
i added unit tests for hashing and comparing

### If needed, did you change documentation(`README.md`, `ARCHITECTURE.md`, etc) to reflect your changes?

- [x] Is it safe to rollback this change if anything goes wrong?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,4 +31,4 @@ On dev mode you can GET at `/graphql` to get graphql playground
 This can be read at `ARCHITECTURE.md`
 
 ## Migrate
-go run ./migrations/init.go
+To migrate the db run `go run ./migrations/init.g`
```
Comment (Oasixer) [inline README.md:34]: should be .go not .g

**Review by Oasixer — COMMENTED** [review]
approve except for the typo i mentioned

File: README.md (hunk 1 above)
Comment (prairir) [reply README.md]: fixed it

**Review by Oasixer — APPROVED** [review]
yooo looks great

**Pull request merged**